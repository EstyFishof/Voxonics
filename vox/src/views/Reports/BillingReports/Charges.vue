<template>
  <div>
    <div class="top-block">
      <BillingMenu />
    </div>
    <div class="wrap-table">
      <table>
        <thead>
          <tr class="header">
            <th @click="sort('date', 'dataArray', 'dateRef')">Date & Time <img src="@/assets/img/icons/sort-arrow.svg" ref="dateRef" /></th>
            <th @click="sort('connectedService.service.purpose', 'dataArray', 'purposeRef')">Purpose <img src="@/assets/img/icons/sort-arrow.svg" ref="purposeRef" /></th>
            <th @click="sort('v-number', 'dataArray', 'v-numberRef')" class="v-number">V-number <img src="@/assets/img/icons/sort-arrow.svg" ref="v-numberRef" /></th>
            <th @click="sort('connectedService.description', 'dataArray', 'descriptionRef')" class="description">Description <img src="@/assets/img/icons/sort-arrow.svg" ref="descriptionRef" /></th>
            <th @click="sort('completed', 'dataArray', 'completedRef')">Charged <img src="@/assets/img/icons/sort-arrow.svg" ref="completedRef" /></th>
            <th @click="sort('chargeAmount', 'dataArray', 'chargeAmountRef')">Amount <img src="@/assets/img/icons/sort-arrow.svg" ref="chargeAmountRef" /></th>
            <th @click="sort('currency', 'dataArray', 'currencyRef')">Ð¡urrency<img src="@/assets/img/icons/sort-arrow.svg" ref="currencyRef" /></th>
            <th>ID</th>
            <th>Info</th>
          </tr>
        </thead>
        <tbody ref="tbody" v-if="dataArray.length">
          <tr v-for="(item, index) of dataArray" :key="index">
            <td>{{item.date}}</td>
            <td>
              <span v-if="item.connectedService && item.connectedService.service">{{item.connectedService.service.purpose}}</span>
            </td>
            <td class="v-number">{{ item.connectedService.number ?? '' }}</td>
            <td>
              <input v-if="item.connectedService" class="description-input" v-model="item.connectedService.description" :readonly="roleAgent !== 'root'" :style="{outline: roleAgent !== 'root' ? 'none' : null}"/>
            </td>
            <td>
              <span class="charged" :style="{background: item.completed ? '#276899' : 'var(--stroke-arrow)'}">{{item.completed ? 'YES' : 'NO'}}</span>
            </td>
            <td>{{item.chargeAmount}}</td>
            <td>{{item.currency}}</td>
            <td><img src="@/assets/img/icons/copy.svg" /></td>
            <td><img src="@/assets/img/icons/info-charges.svg"/></td>
          </tr>
        </tbody>
      </table>
      <Paginator v-if="getPaginatorCharges.paginator.recordsPerPage" :paginator="getPaginatorCharges.paginator" :methods="{page: 'setChargesPage', peerPage: 'setChargesPerPage'}" perPage="12" />
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex"
// import {BillingGettersApi} from '@/API/getters'
import {clientSort, serverSort} from '@/lib/sort.js'
import BillingMenu from './components/BillingMenu'
import Paginator  from '@/components/Paginator'
import observables from '@/store/observable.mutations'

export default {
  name: "Charges",
  components: {BillingMenu, Paginator},
  data() {
    return {
      dataArray: [],
      sortParams: {
        date: 'desc',
        purpose: 'asc',
        'v-number': 'asc',
        description: 'asc',
        completed: 'asc',
        chargeType: 'asc',
        chargeAmount: 'asc',
        currency: 'asc'
      }
    }
  },
  watch: {
    getGlobalCharges() {
      this.dataArray = JSON.parse(JSON.stringify(this.getGlobalCharges))
    }
  },
  computed: {
    ...mapGetters(["getGlobalCharges", 'getPaginatorCharges', 'roleAgent']),
  },
  methods: {
    fetch(peerPage, page, sortParam, orderParam) {
      console.log(orderParam)
      this.$store.dispatch('setGlobalChanges', {peerPage, page, sort: sortParam, orderParam})
    },
    sort(param, arrayParam, refElement) {
      if(param === 'completed' || param === 'chargeAmount' || param === 'date' || param === 'currency') {
        const dataSort = serverSort(this.fetch, this.$refs[refElement], this.sortParams[param], this.getPaginatorCharges.peerPage, this.getPaginatorCharges.page, param)
        this.sortParams[param] = dataSort.order
        this.$refs[refElement] = dataSort.refArrow
      } else {
        const dataSort = clientSort(param, this[arrayParam], this.$refs[refElement], this.sortParams[param])
        this[arrayParam] = dataSort.array
        this.sortParams[param] = dataSort.order
        this.$refs[refElement] = dataSort.refArrow
      }
    }
  },
  mounted() {
    this.$store.dispatch('setGlobalChanges', {peerPage: 12, page: 1, sort: 'date', orderParam: 'desc'})

    this.$store.subscribe((mutation) => {
      if (observables.paginator.charges.includes(mutation.type)) {
        this.$store.dispatch('setGlobalChanges', {peerPage: this.getPaginatorCharges.peerPage, page: this.getPaginatorCharges.page, sort: 'date', orderParam: 'desc'})
      }
    })
  }
}
</script>

<style scoped lang="scss">
.top-block{
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: relative;

  &::after {
    content: '';
    position: absolute;
    width: 100%;
    height: 2px;
    bottom: 26px;
    background: var(--cdr-border-row);
    border-radius: 8px;
  }
}
.wrap-table {
  table {
    width: 100%;
    border-collapse: separate;
    border-spacing: 0 6px;

    thead {
      .header {
        border: 2px solid var(--light-blue);
        background-color: var(--light-blue);

        th {
          font-size: 14px;
          font-weight: 400;
          height: 44px;
          text-align: left;
          padding-left: 18px;
          cursor: pointer;

          &.v-number {
            width: 13%;
            text-align: center;
            padding-left: 0;
          }
          &.description{
            width: 27%;
          }
        }
      }
    }

    tbody {
      tr {
        border: 2px solid var(--statement-border);
        box-shadow: 0 5px 16px rgba(0, 0, 0, 0.1);
        -moz-box-shadow: 0 5px 16px rgba(0, 0, 0, 0.1);
        -webkit-box-shadow: 0 5px 16px rgba(0, 0, 0, 0.1);

        td {
          font-size: 16px;
          font-weight: 300;
          height: 52px;
          text-align: left;
          padding-left: 18px;
          position: relative;
          color: var(--font-panel);
          border-top: 2px solid var(--statement-border);
          border-bottom: 2px solid var(--statement-border);

          &:first-child {
            border-left: 2px solid var(--statement-border);
          }

          &:last-child {
            border-right: 2px solid var(--statement-border);
          }
          .flag {
            margin: 0 2px;
          }
          &.v-number{
            text-align: center;
          }
          .description-input{
            background: var(--input-bg);
            border-radius: 4px;
            width: 331px;
            height: 24px;
            padding: 2px 13px;
            font-weight: 300;
            font-size: 16px;
            line-height: 19px;
            border: none;
            color: var(--font-panel);
          }
          .charged {
            width: 46px;
            height: 20px;
            display: flex;
            align-items: center;
            justify-content: center;
            border-radius: 6px;
            font-weight: 500;
            font-size: 14px;
            line-height: 16px;
            color: white;
          }

          .cursor {
            cursor: pointer;

            &:hover {
              transform: scale(1.1);
            }

            &:active {
              transform: scale(1);
            }
          }
        }
      }
    }
  }
}
</style>
