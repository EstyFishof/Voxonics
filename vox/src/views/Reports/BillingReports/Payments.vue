<template>
  <div>
    <BillingMenu />
    <div class="wrap-table">
      <table>
        <thead>
          <tr class="header">
            <th @click="sort('dateCreated', 'dateCreatedRef')">Date & Time <img src="@/assets/img/icons/sort-arrow.svg" ref="dateCreatedRef" /></th>
            <th>Method </th>
            <th @click="sort('amount', 'amountRef')">Amount <img src="@/assets/img/icons/sort-arrow.svg" ref="amountRef" /></th>
            <th @click="sort('currency', 'currencyRef')">Сurrency <img src="@/assets/img/icons/sort-arrow.svg" ref="currencyRef" /></th>
            <th @click="sort('TAX', 'TAXRef')">VAT/Tax <img src="@/assets/img/icons/sort-arrow.svg" ref="TAXRef" /></th>
            <th  @click="sort('status', 'statusRef')">Status <img src="@/assets/img/icons/sort-arrow.svg" ref="statusRef" /></th>
            <th @click="sort('paidDate', 'paidDateRef')">Paid at <img src="@/assets/img/icons/sort-arrow.svg" ref="paidDateRef" /></th>
            <th @click="sort('billAmount', 'billAmountRef')">Total Paid <img src="@/assets/img/icons/sort-arrow.svg" ref="billAmountRef"/></th>
            <th @click="sort('invoiceCurrency', 'invoiceCurrencyRef')">Сurrency <img src="@/assets/img/icons/sort-arrow.svg" ref="invoiceCurrencyRef" /></th>
            <th class="invoice">Invoice</th>
            <th class="invoice-id question-id">
              ID
              <img class="question" src="@/assets/img/icons/question.svg" />
            </th>
            <th class="receipt">Receipt</th>
            <th class="question-id">
              ID
              <img class="question" src="@/assets/img/icons/question.svg" />
            </th>
          </tr>
        </thead>
        <tbody ref="tbody" v-if="dataPayments.length">
          <tr v-for="(item, index) in dataPayments" :key="index">
            <td>{{item.dateCreated}}</td>
            <td></td>
            <td>{{item.amount}}</td>
            <td>{{item.currency}}</td>
            <td>{{item.tax}}%</td>
            <td>
              <img class="status-icon" :src="getIconPayment(item.status)" :content="item.status" width="20" v-tippy="{ placement : 'top',  arrow: true, delay : 800, duration: 500 }" />
            </td>
            <td>{{item.paidDate}}</td>
            <td>{{item.billAmount}}</td>
            <td>{{item.invoiceCurrency}}</td>
            <td class="invoice"><img src="@/assets/img/icons/download.svg" /></td>
            <td class="invoice-id question-id">
              <img src="@/assets/img/icons/copy.svg" />
            </td>
            <td class="receipt">
              <img v-if="item.status === 'success'" src="@/assets/img/icons/download.svg" />
              <div v-if="item.status === 'pending'" class="upload" @click="trigger()" :class="{uploaded: file}">
                <span class="upload-text">{{file ? 'Confirmation Uploaded' : 'Upload confirmation'}}</span>
                <img class="upload-img" :src="file ? successStatusPayment : uploadImg"/>
                <input @change="uploadFile($event)" type="file" class="upload-input" ref="uploadInput">
              </div>
            </td>
            <td>
              <img v-if="item.status === 'success'" src="@/assets/img/icons/copy.svg" />
            </td>
          </tr>
        </tbody>
      </table>
      <Paginator v-if="getPaginatorPayments.paginator.recordsPerPage" :paginator="getPaginatorPayments.paginator" :methods="{page: 'setPaymentsPage', peerPage: 'setPaymentsPerPage'}" perPage="12" />
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex"
// import {BillingGettersApi} from '@/API/getters'
import {serverSort} from '@/lib/sort.js'
import BillingMenu from './components/BillingMenu'
import Paginator  from '@/components/Paginator'
import observables from '@/store/observable.mutations'
import redStatusPayment from '@/assets/img/icons/red-status-shadow.svg'
import successStatusPayment from '@/assets/img/icons/report-success-shadow.svg'
import pendingStatusPayment from '@/assets/img/icons/pending-payment-shadow.svg'
import uploadImg from '@/assets/img/icons/upload.svg'

export default {
  name: "Payments",
  components: {BillingMenu, Paginator},
  data() {
    return {
      dataPayments: [],
      sortParam: {
        dateCreated: 'desc',
        amount: 'desc',
        currency: 'desc',
        TAX: 'desc',
        paymentDescription: 'desc',
        paidDate: 'desc',
        billAmount: 'desc',
        invoiceCurrency: 'desc',
        invoiceNumber: 'desc',
      },
      redStatusPayment: redStatusPayment,
      successStatusPayment: successStatusPayment,
      pendingStatusPayment: pendingStatusPayment,
      uploadImg: uploadImg,
      file: null
    }
  },
  watch: {
    getGlobalPayments() {
      this.dataPayments = JSON.parse(JSON.stringify(this.getGlobalPayments))
    }
  },
  computed: {
    ...mapGetters(["getGlobalPayments", 'getPaginatorPayments']),
  },
  methods: {
    sort(param, refElement) {
      const dataSort = serverSort(this.fetch, this.$refs[refElement], this.sortParam[param], this.getPaginatorPayments.peerPage, this.getPaginatorPayments.page, param)
      this.sortParam[param] = dataSort.order
      this.$refs[refElement] = dataSort.refArrow
    },
    fetch(peerPage, page, param, order) {
      this.$store.dispatch('setGlobalPayments', {peerPage, page, param, order})
    },
    getIconPayment(status) {
      switch (status) {
        case 'success':
          return this.successStatusPayment
        case 'pending':
          return this.pendingStatusPayment
        default:
          return this.redStatusPayment
      }
    },
    trigger() {
      this.$refs.uploadInput[0].click()
    },
    uploadFile(event) {
      this.file = event.target.files[0]
    },
  },
  mounted() {
    this.$store.dispatch('setGlobalPayments', {peerPage: 12, page: 1})

    this.$store.subscribe((mutation) => {
      if (observables.paginator.payments.includes(mutation.type)) {
        this.$store.dispatch('setGlobalPayments', {peerPage: this.getPaginatorPayments.peerPage, page: this.getPaginatorPayments.page})
      }
    })
  }
}
</script>

<style scoped lang="scss">
.wrap-table {
  table {
    width: 100%;
    border-collapse: separate;
    border-spacing: 0 6px;

    thead {
      .header {
        border: 2px solid var(--light-blue);
        background-color: var(--light-blue);

        th {
          font-size: 14px;
          font-weight: 400;
          height: 44px;
          text-align: left;
          padding-left: 18px;
          cursor: pointer;
          position: relative;

          &.invoice {
            &::before {
              content: '';
              position: absolute;
              width: 2px;
              height: 38px;
              top: 3px;
              left: 0;
              background: var(--paginator-bg);
              border-radius: 20px;
            }
          }

          &.invoice, &.receipt {
            width: 70px;
            padding-left: 22px;
          }

          &.invoice-id {
            padding-right: 18px;
            text-align: center;
            &::after {
              content: '';
              position: absolute;
              width: 2px;
              height: 38px;
              top: 3px;
              right: 0;
              background: var(--paginator-bg);
              border-radius: 20px;
            }
          }

          &.question-id {
            position: relative;
            padding-right: 28px;
            padding-left: 25px;
            text-align: center;
            width: 67px;
            .question {
              position: absolute;
              top: 12px;
              right: 20px;
            }
          }
        }
      }
    }

    tbody {
      tr {
        border: 2px solid var(--statement-border);
        box-shadow: 0 5px 16px rgba(0, 0, 0, 0.1);
        -moz-box-shadow: 0 5px 16px rgba(0, 0, 0, 0.1);
        -webkit-box-shadow: 0 5px 16px rgba(0, 0, 0, 0.1);

        td {
          font-size: 16px;
          font-weight: 300;
          height: 52px;
          text-align: left;
          padding-left: 18px;
          position: relative;
          color: var(--font-panel);
          border-top: 2px solid var(--statement-border);
          border-bottom: 2px solid var(--statement-border);
          position: relative;

          &:first-child {
            border-left: 2px solid var(--statement-border);
          }

          &:last-child {
            border-right: 2px solid var(--statement-border);
          }
          .status-icon {
            margin-top: 6px;
          }

          &.invoice {
            padding-left: 0;
            text-align: center;
            &::before {
              content: '';
              position: absolute;
              width: 2px;
              height: 38px;
              top: 3px;
              left: 0;
              background: var(--scenario-option-bg);
              border-radius: 20px;
            }
          }

          &.invoice-id {
            padding-left: 0;
            text-align: center;
            &::after {
              content: '';
              position: absolute;
              width: 2px;
              height: 38px;
              top: 3px;
              right: 0;
              background: var(--scenario-option-bg);
              border-radius: 20px;
            }
          }

          &.receipt {
            padding-left: 0;
            text-align: center;
          }

          .flag {
            margin: 0 2px;
          }

          .cursor {
            cursor: pointer;

            &:hover {
              transform: scale(1.1);
            }

            &:active {
              transform: scale(1);
            }
          }

          .upload {
            width: 101px;
            height: 34px;
            padding: 3px 6px 4px 7px;
            position: absolute;
            top: 7px;
            left: 23px;
            display: flex;
            align-items: center;
            justify-content: space-between;
            border: 1px solid #276899;
            box-sizing: border-box;
            border-radius: 4px;
            cursor: pointer;

            .upload-text {
              font-weight: normal;
              font-size: 11px;
              line-height: 14px;
              color: var(--scenario-tab-color);
              text-align: left;
            }

            .upload-input {
              display: none;
            }
          }
        }
      }
    }
  }
}
</style>
