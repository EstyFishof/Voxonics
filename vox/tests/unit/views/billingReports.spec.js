import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import Statement from '@/views/Reports/BillingReports/Statement.vue';
import Payments from '@/views/Reports/BillingReports/Payments.vue';
import Charges from '@/views/Reports/BillingReports/Charges.vue';
import mock from './__mock__/billing';
import VueTippy, { TippyComponent } from "vue-tippy";


const localVue = createLocalVue();
localVue.use(Vuex);
const router = new VueRouter();
localVue.use(Vuex);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);

window.env = {
    BASE_URL: 'http://168.119.92.172/'
};

const paginator = {peerPage: 10, page: 1, paginator: {count: 50, recordsPerPage: 5}};

describe('Statement', () => {

    const store = new Vuex.Store({
        getters: {
            getStatement: () => mock.statement,
            getLastPayments: () => mock.getLastPayments,
            getLastCharges: () => mock.getLastCharges,
            theme: () => 'dark'
        }
    });

    const wrapper = shallowMount(Statement, { localVue, router, store });
    it('it should  Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('getIconPayment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'getIconPayment');
        await wrapper.vm.getIconPayment('success');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('getIconPayment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'getIconPayment');
        await wrapper.vm.getIconPayment('pending');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('getIconPayment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'getIconPayment');
        await wrapper.vm.getIconPayment('');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('computed lastRefill', () => {
        Statement.computed.lastRefill.call(wrapper.vm);
    });
    it('mounted component', async () => {
        await Statement.mounted.call(wrapper.vm);
    });
});

describe('Payments', () => {

    const store = new Vuex.Store({
        getters: {
            getStatement: () => mock.statement,
            getLastPayments: () => mock.getLastPayments,
            getLastCharges: () => mock.getLastCharges,
            getPaginatorPayments: () => paginator,
            theme: () => 'dark'
        }
    });

    const wrapper = shallowMount(Payments, { localVue, router, store });
    it('it should  Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('getIconPayment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'getIconPayment');
        await wrapper.vm.getIconPayment('success');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('getIconPayment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'getIconPayment');
        await wrapper.vm.getIconPayment('pending');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('getIconPayment toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'getIconPayment');
        await wrapper.vm.getIconPayment('');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('mounted component', async () => {
        await Payments.mounted.call(wrapper.vm);
    });
});

describe('Charges', () => {

    const store = new Vuex.Store({
        getters: {
            getStatement: () => mock.statement,
            getLastPayments: () => mock.getLastPayments,
            getLastCharges: () => mock.getLastCharges,
            getPaginatorCharges: () => paginator,
            theme: () => 'dark'
        }
    });

    const wrapper = shallowMount(Charges, { localVue, router, store });
    it('it should  Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('sort toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'sort');
        await wrapper.vm.sort('date');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('sort toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'sort');
        await wrapper.vm.sort('purpose');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('sort toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'sort');
        await wrapper.vm.sort('country');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('sort toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'sort');
        await wrapper.vm.sort('description');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('sort toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'sort');
        await wrapper.vm.sort('completed');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('sort toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'sort');
        await wrapper.vm.sort('chargeType');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('sort toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'sort');
        await wrapper.vm.sort('chargeAmount');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });

    it('mounted component', async () => {
        await Charges.mounted.call(wrapper.vm);
    });
});