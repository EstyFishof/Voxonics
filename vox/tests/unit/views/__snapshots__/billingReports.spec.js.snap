// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charges Check snapshot 1`] = `
<div>
  <div
    class="top-block"
  >
    <billingmenu-stub />
  </div>
   
  <div
    class="wrap-table"
  >
    <table>
      <thead>
        <tr
          class="header"
        >
          <th>
            Date & Time 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            Purpose 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th
            class="v-number"
          >
            V-number 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th
            class="description"
          >
            Description 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            Charged 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            Amount 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            Сurrency
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            ID
          </th>
           
          <th>
            Info
          </th>
        </tr>
      </thead>
       
      <!---->
    </table>
     
    <paginator-stub
      methods="[object Object]"
      paginator="[object Object]"
    />
  </div>
</div>
`;

exports[`Payments Check snapshot 1`] = `
<div>
  <billingmenu-stub />
   
  <div
    class="wrap-table"
  >
    <table>
      <thead>
        <tr
          class="header"
        >
          <th>
            Date & Time 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            Method 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            Amount 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            Сurrency 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            VAT/Tax 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            Status 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            Paid at 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            Total Paid 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th>
            Сurrency 
            <img
              src="@/assets/img/icons/sort-arrow.svg"
            />
          </th>
           
          <th
            class="invoice"
          >
            Invoice
          </th>
           
          <th
            class="invoice-id question-id"
          >
            
            ID
            
            <img
              class="question"
              src="@/assets/img/icons/question.svg"
            />
          </th>
           
          <th>
            Receipt
          </th>
           
          <th
            class="question-id"
          >
            
            ID
            
            <img
              class="question"
              src="@/assets/img/icons/question.svg"
            />
          </th>
        </tr>
      </thead>
       
      <!---->
    </table>
     
    <paginator-stub
      methods="[object Object]"
      paginator="[object Object]"
    />
  </div>
</div>
`;

exports[`Statement Check snapshot 1`] = `
<div>
  <billingmenu-stub />
   
  <div
    class="statement-block"
  >
    <div
      class="left-sector"
    >
      <div
        class="balance-block"
      >
        <div
          class="company-name"
        >
          
          
        
        </div>
         
        <span
          class="deposit-title"
        >
          Your deposit balance
        </span>
         
        <span
          class="deposit-cost"
        >
          400.00    USD
        </span>
         
        <span
          class="refill"
        >
          Last refill   USD
        </span>
         
        <span
          class="deposit-btn"
        >
          Account topup
        </span>
      </div>
       
      <!---->
       
      <div
        class="last-month"
      >
        <span
          class="last-month-title"
        >
          Last month spent
        </span>
         
        <span
          class="last-month-value"
        >
          $ 
        </span>
      </div>
    </div>
     
    <div
      class="right-sector"
    >
      <ul
        class="common-data"
      >
        <li
          class="common-data-item"
        >
          <span
            class="common-data-title"
          >
            Next 10 day’s charges
          </span>
           
          <div
            class="common-data-price"
          >
            
            $ 
            
            <img
              alt="changes"
              src="@/assets/img/icons/next-charges.svg"
            />
          </div>
        </li>
         
        <li
          class="common-data-item"
        >
          <span
            class="common-data-title"
          >
            Avarage usage per day
          </span>
           
          <div
            class="common-data-price"
          >
            
            $ 
            
            <img
              alt="changes"
              src="@/assets/img/icons/avarage-usage.svg"
            />
             
            <!---->
          </div>
        </li>
         
        <li
          class="common-data-item"
        >
          <span
            class="common-data-title"
          >
            Agents
          </span>
           
          <div
            class="common-data-price common-data-counter"
          >
            
            
            
            <img
              alt="changes"
              src="@/assets/img/icons/agents-statement.svg"
            />
          </div>
        </li>
         
        <li
          class="common-data-item"
        >
          <span
            class="common-data-title"
          >
            V-Numbers
          </span>
           
          <div
            class="common-data-price common-data-counter"
          >
            
            
            
            <img
              alt="changes"
              src="@/assets/img/icons/v-numbers-statement.svg"
            />
          </div>
        </li>
      </ul>
       
      <div
        class="tables-block"
      >
        <div
          class="tables-block-top"
        >
          <div
            class="table-name"
          >
            <img
              src="@/assets/img/icons/payments-white.svg"
            />
             
            <!---->
            
            Last payments
          
          </div>
           
          <router-link-stub
            ariacurrentvalue="page"
            class="tables-block-top-btn"
            event="click"
            tag="a"
            to="/reports/payments"
          >
            
            View all
            
            <img
              src="@/assets/img/icons/statement-view-all.svg"
            />
             
            <!---->
          </router-link-stub>
        </div>
         
        <div
          class="wrap-table"
        >
          <table>
            <thead>
              <tr
                class="header"
              >
                <th>
                  
                Invoice #
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                <th>
                  
                Issued
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                <th>
                  
                Method
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                <th>
                  
                Status
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                <th>
                  
                Total Paid
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                <th>
                  
                Currency
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                <th>
                  
                Paid at
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                 
                <th>
                  Invoice
                </th>
                 
                <th>
                  Receipt
                </th>
              </tr>
            </thead>
             
            <tbody>
              <tr>
                <td>
                  
                </td>
                 
                <td>
                  
                </td>
                 
                <td />
                 
                <td
                  class="status"
                >
                  <img
                    content="success"
                    src=""
                    tabindex="0"
                  />
                </td>
                 
                <td>
                  
                </td>
                 
                <td>
                  
                </td>
                 
                <td>
                  
                </td>
                 
                <td>
                  <img
                    src="@/assets/img/icons/download.svg"
                  />
                </td>
                 
                <td>
                  <img
                    src="@/assets/img/icons/download.svg"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
       
      <div
        class="tables-block"
      >
        <div
          class="tables-block-top"
        >
          <div
            class="table-name"
          >
            <img
              src="@/assets/img/icons/charges-white.svg"
            />
             
            <!---->
            
            Future charges
          
          </div>
           
          <router-link-stub
            ariacurrentvalue="page"
            class="tables-block-top-btn"
            event="click"
            tag="a"
            to="/reports/charges"
          >
            
            View all
            
            <img
              src="@/assets/img/icons/statement-view-all.svg"
            />
             
            <!---->
          </router-link-stub>
        </div>
         
        <div
          class="wrap-table"
        >
          <table>
            <thead>
              <tr
                class="header"
              >
                <th>
                  
                Issued
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                <th>
                  
                Purpose
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                <th>
                  
                Country
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                <th>
                  
                V-number
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                <th>
                  
                Description
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                <th>
                  
                Charge
                
                  <img
                    src="@/assets/img/icons/sort-arrow.svg"
                  />
                   
                  <img
                    src="@/assets/img/icons/sort-arrow-white.png"
                    style="display: none;"
                  />
                </th>
                 
                <th>
                  ID
                </th>
              </tr>
            </thead>
             
            <tbody>
              <tr>
                <td>
                  
                </td>
                 
                <td>
                  
                </td>
                 
                <td>
                  
                </td>
                 
                <td />
                 
                <td
                  class="status"
                >
                  
                </td>
                 
                <td>
                  
                </td>
                 
                <td>
                  <img
                    src="@/assets/img/icons/copy.svg"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
